src/components/sectionList/listActions/DeleteAction.tsx

```diff
@@ -0,0 +1,173 @@
+import { useAlert } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    ButtonStrip,
+    CircularLoader,
+    IconDelete16,
+    MenuItem,
+    Modal,
+    ModalActions,
+    ModalContent,
+    ModalTitle,
+    NoticeBox,
+} from '@dhis2/ui'
+import React, { useState } from 'react'
+import {
+    BaseListModel,
+    useSchemaFromHandle,
+    useDeleteModelMutation,
+} from '../../../lib'
+import classes from './DeleteAction.module.css'
+
+export function DeleteAction({
+    disabled,
+    model,
+    modelType,
+    onCancel,
+    onDeleteSuccess,
+}: {
+    disabled: boolean
+    model: BaseListModel
+    modelType: string
+    onCancel: () => void
+    onDeleteSuccess: () => void
+}) {
+    const [showConfirmationDialog, setShowConfirmationDialog] = useState(false)
+    const deleteAndClose = () => {
+        setShowConfirmationDialog(false)
+        onDeleteSuccess()
+    }
+    const closeAndCancel = () => {
+        setShowConfirmationDialog(false)
+        onCancel()
+    }
+
+    return (
+        <>
+            <MenuItem
+                dense
+                disabled={disabled}
+                label={i18n.t('Delete')}
+                icon={<IconDelete16 />}
+                onClick={() => setShowConfirmationDialog(true)}
+            />
+
+            {showConfirmationDialog && (
+                <ConfirmationDialog
+                    model={model}
+                    modelType={modelType}
+                    onDeleteSuccess={deleteAndClose}
+                    onCancel={closeAndCancel}
+                />
+            )}
+        </>
+    )
+}
+
+function ConfirmationDialog({
+    model,
+    modelType,
+    onCancel,
+    onDeleteSuccess,
+}: {
+    model: BaseListModel
+    modelType: string
+    onCancel: () => void
+    onDeleteSuccess: () => void
+}) {
+    const schema = useSchemaFromHandle()
+    const deleteModelMutation = useDeleteModelMutation(schema, {
+        onSuccess: () => {
+            showDeletionSuccess()
+            onDeleteSuccess()
+        },
+    })
+
+    const { show: showDeletionSuccess } = useAlert(
+        () =>
+            i18n.t('Successfully deleted {{modelType}} "{{displayName}}"', {
+                displayName: model.displayName,
+                modelType,
+            }),
+        { success: true }
+    )
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const errorReports = (deleteModelMutation.error as any)?.details?.response
```

Comment:
I think we should look into having an `ErrorReport`-component that handled this stuff, and had proper types for this.

Reply:
Agree with the component! Added a comment about something minor on the PR, besides that I think it's ready to be merged.